### PR TITLE
feat(keyboard-shortcut): add component Al 2134

### PR DESCRIPTION
**Pull request by ianaa**
The component code and unit tests copied with slight modifications from editor (credit goes to @manuatarchilogic)
Based on @bnjm's advice, I put `@archilogic/toolbox` as a devDependency, and it is included in the library build. 
That made me notice that some newer dependencies weren't being externalised in the vite config, so fixed that too.

### CODE REVIEW

File: src/stories/KeyboardShortcut.stories.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import type { Meta, StoryObj } from '@storybook/vue3'
+import type { ComponentProps } from 'vue-component-type-helpers'
+import { AKeyboardShortcut } from '../components'
+
+type KeyboardShortcutProps = ComponentProps<typeof AKeyboardShortcut>
+
+/**
+ */
```
Comment (manuatarchilogic) [inline src/stories/KeyboardShortcut.stories.ts]:
```suggestion
```

File: src/stories/KeyboardShortcut.stories.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  components: { AKeyboardShortcut },
+  setup() {
+    return { args }
+  },
+  template: `
+  <AKeyboardShortcut v-bind="args"></AKeyboardShortcut>`
+})
+
+/**
+ */
```
Comment (manuatarchilogic) [inline src/stories/KeyboardShortcut.stories.ts]:
```suggestion
```

**Review by manuatarchilogic — APPROVED** [review]
nice :)

all uses of keyboard shortcuts in Editor except 1 are inside `<template #extra>` of `<a-app-menu-item>`. We could have a `#shortcuts` slot in `<a-app-menu-item>` that automatically creates a `<a-keyboard-shortcut>`

**Review by bnjm — APPROVED** [review]
Looks great! Might be worth creating a pre-release to make sure we didn't break anything with the dependency changes?

File: src/components/__tests__/KeyboardShortcut.test.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    })
+  })
+
+  it('renders a "ctrl" modifier key', () => {
+    const { getByText } = render(KeyboardShortcut, {
+      props: { shortcut: { modifiers: ['ctrl'] } }
+    })
+
+    expect(getByText('Ctrl')).toBeTruthy()
+  })
```
Comment (romiekos) [inline src/components/__tests__/KeyboardShortcut.test.ts:39]: That repetitive block could be transformed into a function and then we can use an array of tests.

```
const testModifierKey = (key, keyText) => {
  it(`renders a "${key}" modifier key`, () => {
    const { getByText } = render(KeyboardShortcut, {
      props: { shortcut: { modifiers: [key] } }
    })

    expect(getByText(keyText)).toBeTruthy()
  })
}
```

**Review by romiekos — APPROVED** [review]

**Comment by manuatarchilogic** [comment]
> nice :)
> 
> all uses of keyboard shortcuts in Editor except 1 are inside `<template #extra>` of `<a-app-menu-item>`. We could have a `#shortcuts` slot in `<a-app-menu-item>` that automatically creates a `<a-keyboard-shortcut>`

I mean, a `shortcuts` prop obviously, not a slot

**Comment by ianaa** [comment]
> > nice :)
> > all uses of keyboard shortcuts in Editor except 1 are inside `<template #extra>` of `<a-app-menu-item>`. We could have a `#shortcuts` slot in `<a-app-menu-item>` that automatically creates a `<a-keyboard-shortcut>`
> 
> I mean, a `shortcuts` prop obviously, not a slot

yep,  I will make a follow up with this prop for app-menu-item and for tooltips too

File: src/components/__tests__/KeyboardShortcut.test.ts (hunk 3 above)
Comment (ianaa) [reply src/components/__tests__/KeyboardShortcut.test.ts:39]: oh but I like my repetitive tests... 🥺

File: src/stories/KeyboardShortcut.stories.ts (hunk 1 above)
Comment (ianaa) [reply src/stories/KeyboardShortcut.stories.ts]: I meant to add some docs there 🤦‍♀️ , hang on

**Comment by ianaa** [comment]
> Looks great! Might be worth creating a pre-release to make sure we didn't break anything with the dependency changes?

I think it's ok to test this as a normal release and make a fix if needed - the process would be the same with a prerelease.
The package would not get automatically updated to a potentially broken version anywhere in our apps.

**Pull request merged**